homeassistant/components/switchbot/sensor.py

```diff
@@ -66,12 +77,22 @@ async def async_setup_entry(
                 entry.data[CONF_MAC],
                 entry.data[CONF_NAME],
             )
-            for sensor in coordinator.data[entry.unique_id]["data"]
+            for sensor in flatten_sensors_data(
+                coordinator.data[entry.unique_id]["data"]
+            )
             if sensor in SENSOR_TYPES
         ]
     )
 
 
+def flatten_sensors_data(sensors):
+    """Deconstruct SwitchBot library temp object C/Fº readings from object."""
+    if "temp" in sensors:
+        sensors["temperature_celsius"] = sensors["temp"]["c"]
```

Comment:
```suggestion
        sensors["temperature"] = sensors["temp"]["c"]
```

The unit is already defined above

Reply:
Updated in a new commit, because the dictionary for sensor types needed to change along with this